### ISSUE TITLE
loadFull() replaces groups

### ISSUE THREAD

**Issue by othlu** [mongator/mongator#8]
Calling `loadFull()` in an embedded document will completely replace the group in the parent model. This means, that any further operations in the embedded document, will not affect the embeddeds in the parent model.

The changes were partially introduced with f121af33189bc8df517189b5888d2331438f5ecf

Looking at lines 73-81 in DocumentFields.php.twig and assuming an embedded document:
- if the field is not present, mongator loads the embedded fields, and replaces the group in the parent document (73-77)
- now the current object/embedded document ($this) is no longer attached to the parent model
- finally, lines 79-81 will never kick in after a call to loadFull in an embedded document, as the embedded itself was never changed.

**Comment by dripolles** [comment]
Hey, thanks for the detailed report :+1: 

I'm still trying to figure out which conditions can lead to this behaviour, but it might be related to this line:

https://github.com/mongator/mongator/blob/master/src/Mongator/Document/Document.php#L265

If the embedded fields do not get incluided in $this->fieldsModified, that would be a simple explanation for the overwrite. I'm sure it can be fixed, but I can't think about a "quick fix" right now.

We're in a bit of a rush here with some internal project, so I can't investigate this further at this moment. I will try to find some time for it, but unfortunately I can't promise it at this time.

At least I hope this helps you fix or work around the issue.

**Comment by othlu** [comment]
I suspect the problem sitting deeper in the loadFull conception. Especially if dealing with deeper nested documents, if I am correct, the `$this->fieldsModified` only catches changes in the parent model and not the children. A simple workaround is to trigger loadFull before altering not loaded fields.

**Comment by dripolles** [comment]
Yes, `fieldsModified` only "sees" changes in the parent model. I guess it would be possible to use it only for "scalar" fields and call `isModified` for embeddeds (I'm not sure about references).

Calling `loadFull` as you said is probably the easiest workaround without getting deeper into the issue.
